src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -2073,22 +2354,74 @@ public BeforePublishEvent(System.IServiceProvider services, ApplicationModel.Dis
         public System.IServiceProvider Services { get { throw null; } }
     }
 
+    [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPUBLISHERS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public enum CompletionState
+    {
+        InProgress = 0,
+        Completed = 1,
+        CompletedWithWarning = 2,
+        CompletedWithError = 3
+    }
+
+    [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPUBLISHERS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public partial class ContainerBuildOptions
+    {
+        public ContainerImageFormat? ImageFormat { get { throw null; } init { } }
+
+        public string? OutputPath { get { throw null; } init { } }
+
+        public ContainerTargetPlatform? TargetPlatform { get { throw null; } init { } }
+    }
+
+    [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPUBLISHERS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public enum ContainerImageFormat
+    {
+        Docker = 0,
+        Oci = 1
+    }
+
+    [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPUBLISHERS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public enum ContainerTargetPlatform
+    {
+        LinuxAmd64 = 0,
+        LinuxArm64 = 1,
+        LinuxArm = 2,
+        Linux386 = 3,
+        WindowsAmd64 = 4,
+        WindowsArm64 = 5
+    }
+
     public partial interface IDistributedApplicationPublisher
     {
         System.Threading.Tasks.Task PublishAsync(ApplicationModel.DistributedApplicationModel model, System.Threading.CancellationToken cancellationToken);
     }
 
     [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPUBLISHERS001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
-    public partial interface IPublishingActivityProgressReporter
+    public partial interface IPublishingActivityReporter
+    {
+        System.Threading.Tasks.Task CompletePublishAsync(string? completionMessage = null, CompletionState? completionState = null, bool isDeploy = false, System.Threading.CancellationToken cancellationToken = default);
```

Comment:
Instead, consider `CompletePipelineAsync` as this API is used in deploying as well.

Reply:
Should just be `CompleteAsync`

Reply:
https://github.com/dotnet/aspire/pull/10689

Reply:
I think renaming it to `CompleteAsync` makes the `isDeploy` parameter even more awkward.

Can the `IPublishingActivityReporter` know ambiently if it is a deploy operation or not?

Reply:
IMO, `CompleteAsync` is too vague considering there are 3 distinct concepts that can be completed here (the publish/deploy itself, steps, and tasks).

> Can the IPublishingActivityReporter know ambiently if it is a deploy operation or not?

Nope, we'd have to modify this API to take `PublishingOptions` as input if we wanted to determine this without having an `isDeploy` argument.